includes/admin/class-amp-story-templates.php

```diff
@@ -0,0 +1,216 @@
+<?php
+/**
+ * AMP Story Templates handler.
+ *
+ * @package AMP
+ * @since 1.?
+ */
+
+/**
+ * Class AMP_Story_Templates
+ */
+class AMP_Story_Templates {
+
+	/**
+	 * Current version of the templates.
+	 *
+	 * @var string
+	 */
+	const STORY_TEMPLATES_VERSION = '0.1';
+
+	/**
+	 * Slug for templates' taxonomy.
+	 */
+	const TEMPLATES_TAXONOMY = 'amp_template';
+
+	/**
+	 * Slug for templates term.
+	 */
+	const TEMPLATES_TERM = 'story-template';
+
+	/**
+	 * Init.
+	 */
+	public function init() {
+		if ( ! post_type_exists( 'amp_story' ) ) {
+			return;
+		}
+
+		add_filter( 'rest_wp_block_query', array( $this, 'filter_rest_wp_block_query' ), 10, 2 );
+		add_action( 'save_post_wp_block', array( $this, 'flag_template_as_modified' ) );
+
+		$this->register_taxonomy();
+		$this->maybe_import_story_templates();
+	}
+
+	/**
+	 * Import story templates if they haven't been imported previously.
+	 */
+	public function maybe_import_story_templates() {
+		if ( self::STORY_TEMPLATES_VERSION === AMP_Options_Manager::get_option( 'story_templates_version' ) ) {
+			return;
+		}
+		$this->import_story_templates();
+		// @todo This is commented out for testing purposes until the PR gets ready.
+		// AMP_Options_Manager::update_option( 'story_templates_version', self::STORY_TEMPLATES_VERSION );
+	}
+
+	/**
+	 * Import logic for story templates.
+	 */
+	public function import_story_templates() {
+		$templates = self::get_story_templates();
```

Comment:
Idea: What if each story template also included a path to a featured image which could be associated with the `wp_block` post type? This would be like a `screenshot.php` for a theme, where it could show it being populated with content. But the actual block that gets inserted would have the placeholders that users would have to provide content for? This would only apply to templates that are inserted by the AMP plugin, naturally, unless the user is able to set a featured image for `wp_block` post type generally.

Reply:
That's something that @amedina also suggested and that is logged as an idea under this issue:
https://github.com/ampproject/amp-wp/issues/2061

The problem mentioned with this is that the templates are not static, meaning that if the user changes the template then it wouldn't be accurate anymore.

We could stop showing it after the template is changed but this could be unexpected / feel inconsistent.

Maybe we can reiterate under #2061 which is an issue specifically for the template preview?